src/Sandbox.php

```diff
@@ -38,87 +39,55 @@ class Sandbox
     private ?int $expirationAfterHours = null;
     private ?string $deletedTimestamp = null;
 
-    public function __construct(?array $sandbox = null)
-    {
-        if (!empty($sandbox['id'])) {
-            $this->setId((string) $sandbox['id']);
-        }
-        if (!empty($sandbox['projectId'])) {
-            $this->setProjectId((string) $sandbox['projectId']);
-        }
-        if (!empty($sandbox['tokenId'])) {
-            $this->setTokenId((string) $sandbox['tokenId']);
-        }
-        if (!empty($sandbox['configurationId'])) {
-            $this->setConfigurationId((string) $sandbox['configurationId']);
-        }
-        if (!empty($sandbox['physicalId'])) {
-            $this->setPhysicalId($sandbox['physicalId']);
-        }
 
-        if (!empty($sandbox['type'])) {
-            $this->setType($sandbox['type']);
+    public static function fromArray(array $in): self
+    {
+        $sandbox = new Sandbox();
+        if (!isset($in['id'])) {
```

Comment:
no asi bych to napsal jako
foreach ($requiredProps as $prop) {
if (!isset($in[$prop])) throw InvalidApiResponseException(....
}

:)

Reply:
Provedu 👍